contracts/StateChannel.sol

```diff
@@ -254,8 +257,33 @@ contract StateChannel is Initializable, OwnableUpgradeable {
 
         // transfer the balance to contract
         IERC20(settings.getContractAddress(SQContracts.SQToken)).safeTransferFrom(consumer, address(this), amount);
+
+        channels[channelId].realTotal += amount;
+        channels[channelId].total += amount;
+
+        emit ChannelFund(channelId, channels[channelId].realTotal, channels[channelId].total);
+    }
+
+    function fundRewards(uint256 channelId, uint256 amount) external {
+        require(
+            channels[channelId].status == ChannelStatus.Open && channels[channelId].expiredAt > block.timestamp,
+            'SC003'
+        );
+
+        address consumer = channels[channelId].consumer;
+        if (msg.sender != consumer) {
+            bool isController = IConsumerRegistry(settings.getContractAddress(SQContracts.ConsumerRegistry)).isController(consumer, msg.sender);
+            require(isController, 'SC012');
+        }
+
+        // TODO: transfer the balance to contract
+        address rbAddress = settings.getContractAddress(SQContracts.RewardsBooster);
+        IRewardsBooster(rbAddress).spent(consumer, amount);
```

Comment:
Do we need to limit the `charge amount <= realTotal` ?

Reply:
That's what we debated right? if consumer can not withdraw the over-funded token, and we can refund them. 
I  don't mind if amount > realTotal